### DISCUSSION TITLE
is parsing scala-cli .bsp directory out of scope for metals?

### DISCUSSION THREAD

**Discussion by ag91** [scalameta/metals#5479] [Q&A] [answered]
Hello,

I was trying to run metals on a sample script (.sc file) whose directory was set in this way:

``` sh
scala-cli setup-ide . --dep com.lihaoyi::os-lib:0.9.0
```
And I discovered that metals couldn't find the dependency library,
although the .bsp directory created by scala-cli contained information about the dependency.

I gave a peek into [the code](https://github.com/scalameta/metals/blob/main/metals/src/main/scala/scala/meta/internal/metals/scalacli/ScalaCli.scala#L424)  and I realized metals only writes .bsp data.

The workaround was to inject the dependency via directive.
I was tempted to open an issue, but I thought that maybe somebody can share the rationale behind not using a .bsp directory if present.
Would this indeed be out of scope for metals?

**Comment by tgodzik** [comment, marked as answer]
Thanks for the question, Metals does use `.bsp` but the dependency is not contained there. Those are available  for Metals in .scala-build directory. Trying out your examples doesn't cause any issues. os-lib is available within Metals and can run there.

My workflow was:
```
mkdir test
cd test
echo "@main def run = println(os.pwd)" > main-test.scala
scala-cli setup-ide . --dep com.lihaoyi::os-lib:0.9.0
code .
```
and then everything worked fine in Metals.

Although I would suggest using directives for dependencies since you wouldn't need to add the dependency in the terminal each time.

**Comment by ag91** [comment]
Thanks for the explanation! You are right, the example works fine. I got mislead by an incorrect setup of my editor settings (I use Emacs and my [lsp-mode](https://github.com/emacs-lsp/lsp-mode) workspace was pointing to the parent directory instead of the directory with the .scala-build settings and so metals couldn't find the dependencies). 

Great news!
